app/Result.php

```diff
@@ -2,55 +2,46 @@
 namespace API;
 
 use Illuminate\Database\Eloquent\Model;
-use Illuminate\Http\Request;
 
 
 class Result extends Model
 {
-    protected $response;
 
-    function __construct(Request $request)
+    function __construct()
     {
-    	$this->response = $request;
+    	
     }
 
     /*
-     * This method gets the search result for the query made by the user. 
+     * This method gets the search result for the query made by the user from the faroo search api. 
+     *
+     * @param string     src        type of search made by user. 
+     * @param string     query      query made by user.
+     * @return json response received from the faroo search api.
      *
-     * @param param_type param_name param_description
-     * @param string     src        type of search made by user 
-
-     * @return php array 
      */
-    public function getFarooResponse($src)
+    public function getFarooResponse($src,$query)
     {
-    	// Query made by user.
-    	 $query = $this->response->input('query');
  		
         //formating the url using php function.
-        $data = array('q'=>$query,
-                      'start'=>'1',
-                      'length'=>'10',
-                      'l'=>'en',
-                      'src'=>$src,
-                      'i'=>'false',
-                      'f'=>'json');
-
-        $rawUrl = http_build_query($data);
+        $data = ['q'=>$query,
```

Comment:
makes it more readable -
  	 [
		 'q'=>$query,
                  'start'=>'1',
                  'length'=>'10',
                  'l'=>'en',
                  'src'=>$src,
                  'i'=>'false',
                  'f'=>'json'
                ];